src/BuildingBlocks/Tutor.BuildingBlocks.Infrastructure/Database/CrudDatabaseRepository.cs

```diff
@@ -48,6 +48,9 @@ public List<TEntity> BulkCreate(List<TEntity> entities)
 
     public TEntity Update(TEntity entity)
     {
+        var existingEntity = _dbSet.FirstOrDefault(e => e.Id == entity.Id);
```

Comment:
My issue with this change is that it makes 3 database calls (SELECT, DELETE, CREATE) for all update requests (not just the ones added with this PR). Remove this change.

We see an interesting issue here since we are partially using Postgres for its SQL properties, and partially using it to store an aggregate structure that is suitable for a NoSQL context...

I recommend treating the Example as an embedded value object with maybe a natural key "Code" (e.g., "Entitet Bloga" and the "Description"). We can use the "Code" for when we need to find an example (e.g., for event sourcing purposes).